src/create-with-theme.test.js

```diff
@@ -23,34 +22,34 @@ test('should pass the default value of the context', (t) => {
   const theme = {};
   const context = createReactContext(theme);
   const WithTheme = createWithTheme(context)(Comp);
-  const wrapper = mount(<WithTheme />);
+  const { root } = TestRenderer.create(<WithTheme />);
 
-  t.true(wrapper.find('Comp').prop('theme') === theme);
+  t.true(root.findByType(Comp).props.theme === theme);
 });
 
 
 test('should pass the value of the Provider', (t) => {
   const theme = { test: 'test' };
   const context = createReactContext(theme);
   const WithTheme = createWithTheme(context)(Comp);
-  const wrapper = mount((
+  const { root } = TestRenderer.create((
     <context.Provider value={theme}>
       <WithTheme />
     </context.Provider>
   ));
 
-  t.true(wrapper.find('Comp').prop('theme') === theme);
+  t.true(root.findByType(Comp).props.theme === theme);
```

Comment:
I'm testing the wrapped component props. This isn't an internal API

Reply:
It is to the user ....

Reply:
We should test everything same way a user would use it.

Reply:
Why is withTheme an internal API for the user? I'm testing here that withTheme injects the correct theme prop.

Do you maybe mean that I use `createWithTheme` for creating `withTheme` instead of `createTheming`?

Reply:
accessing .props of a component that you wrapped is somewhat internal. Instead you could validate the render output of that combo.

Reply:
Do you mean the test should be more like:
```js
test('should pass the value of the Provider', (t) => {
  const theme = { test: 'test' };
  const context = createReactContext(theme);
  let injectedTheme = null;
  const Comp = (props) => {
    injectedTheme = props.theme;
  };
  const WithTheme = createWithTheme(context)(Comp);
  const { root } = TestRenderer.create((
    <context.Provider value={theme}>
      <WithTheme />
    </context.Provider>
  ));

  // t.true(root.findByType(Comp).props.theme === theme);
  t.true(injectedTheme === theme);
});
```

Reply:
Its better but still not ideal, I really mean we need to actually test the render output, or in case of css generation - generated css.

```js
  const theme = { primary: 'red' };
  const Comp = (props) => {
    <div style={{color: props.theme.primary}}></div>
  };
  const WithTheme = createWithTheme(context)(Comp);
  const { root } = TestRenderer.create((
    <context.Provider value={theme}>
      <WithTheme />
    </context.Provider>
  ));
  // compare the render here, ideally with snapshots
```

Reply:
Yeah, in the case of generated CSS we should test that. Though in this case, we don't generate CSS and actually shouldn't do any DOM testing because the package only depends on react and not ReacDOM specifically.

That's why I went with prop testing.

Reply:
>  Though in this case, we don't generate CSS and actually shouldn't do any DOM testing 

In a real life scenario a theme would be used to either generate CSS or DOM. You can simulate that and use it the way user would do it.

Reply:
Well, the current tests do not assume anything about how the libary is being used and simply assures that the correct prop is injected.